packages/@aws-cdk/aws-appmesh/README.md

```diff
@@ -1,7 +1,194 @@
-## AWS::AppMesh Construct Library
+# AWS AppMesh
 
-This module is part of the [AWS Cloud Development Kit](https://github.com/awslabs/aws-cdk) project.
+This package contains constructs for working with **AWS AppMesh Service**.
 
-```ts
-import appmesh = require('@aws-cdk/aws-appmesh');
+AWS App Mesh is a service mesh based on the [Envoy](https://www.envoyproxy.io/) proxy that makes it easy to monitor and control microservices. App Mesh standardizes how your microservices communicate, giving you end-to-end visibility and helping to ensure high-availability for your applications.
+
+App Mesh gives you consistent visibility and network traffic controls for every microservice in an application.
+
+App Mesh supports microservice applications that use service discovery naming for their components. To use App Mesh, you must have an existing application running on AWS Fargate, Amazon ECS, Amazon EKS, Kubernetes on AWS, or Amazon EC2.
+
+For futher information on **AWS AppMesh** visit the [AWS Docs for AppMesh](https://docs.aws.amazon.com/app-mesh/index.html).
+
+## Creating the Mesh
+
+A service mesh is a logical boundary for network traffic between the services that reside within it.
+
+After you create your service mesh, you can create virtual services, virtual nodes, virtual routers, and routes to distribute traffic between the applications in your mesh.
+
+The following example creates the `AppMesh` service mesh with the default filter of `DROP_ALL`, see [docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-appmesh-mesh-egressfilter.html) here for more info on egress filters.
+
+```typescript
+export const app = new cdk.App();
+
+const mesh = new Mesh(this, 'AppMesh', {
+  name: props.appMeshName,
+});
 ```
+
+The mesh can also be created with the "ALLOW_ALL" egress filter by overwritting the property.
+
+```typescript
+const mesh = new Mesh(this, 'AppMesh', {
+  name: props.appMeshName,
+  meshSpec: {
+    egressFilter: appmesh.MeshFilterType.Allow_All,
+  },
+});
+```
+
+## Exporting the Mesh
+
+The `Mesh` can also be exported to reference in other `Stacks`.
+
+```typescript
+const exportedMesh = mesh.export();
+
+const STACK_2 = new cdk.Stack(app, 'stack-2');
+const importedMesh = appmesh.Mesh.import(STACK_2, 'imported-mesh', exportedMesh);
```

Comment:
This isn't necessary... Within the same application, the instance of the mesh should be passed around, instead of manually doing the `export`/`import` dance.

For cross-application references, this will feel contrived.

Reply:
removed, will show on next commit